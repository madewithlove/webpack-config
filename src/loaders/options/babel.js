export default options => {
    return {
        presets: options.react ? ['react-app', 'stage-0'] : ['env', 'stage-0'],
        plugins: ['transform-decorators-legacy'],
    };
};
