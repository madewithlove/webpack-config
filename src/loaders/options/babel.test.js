import babel from './babel';

let config;

it('can enable React mode', () => {
    config = babel({
        react: true,
        sourcePath: 'foobar',
        loaders: {
            js: 'babel',
        },
    });

    expect(config).toMatchSnapshot();
});

it('can enable HMR', () => {
    config = babel({
        react: true,
        hot: true,
        sourcePath: 'foobar',
        loaders: {
            js: 'babel',
        },
    });

    expect(config).toMatchSnapshot();
});
