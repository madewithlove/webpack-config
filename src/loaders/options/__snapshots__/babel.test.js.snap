// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can enable HMR 1`] = `
Object {
  "plugins": Array [
    "transform-decorators-legacy",
  ],
  "presets": Array [
    "react-app",
    "stage-0",
    "react-hmre",
  ],
}
`;

exports[`can enable React mode 1`] = `
Object {
  "plugins": Array [
    "transform-decorators-legacy",
  ],
  "presets": Array [
    "react-app",
    "stage-0",
  ],
}
`;
